misc/zsh/_khard

```diff
@@ -88,35 +88,35 @@ case $state in
     # input file options
     local -a email_header_input_options template_file_input_options
     email_header_input_options=(
-      '(-i)'{-i+,--input-file=}'[Specify input email header file name or use stdin]:input file:_files'
+      '(-i)'{-i+,--input-file=}'[specify input email header file name or use stdin]:input file:_files'
     )
     template_file_input_options=(
-      '(-i)'{-i+,--input-file=}'[Specify input template file name or use stdin]:input file:_files'
-      '--open-editor[Open text editor after successful creation of new contact from stdin or template]'
+      '(-i)'{-i+,--input-file=}'[specify input template file name or use stdin]:input file:_files'
+      '--open-editor[open text editor after successful creation of new contact from stdin or template]'
     )
     # sort options
     local -a sort_options
     sort_options=(
-      '(-d)'{-d+,--display=}'[display names in contact table by first or last name]:sort by:(first_name last_name)'
+      '(-d)'{-d+,--display=}'[display names in contact table by first or last name]:name:(first_name last_name)'
       '(-g)'{-g,--group-by-addressbook}'[group contacts table by address book]'
       '(-r)'{-r,--reverse}'[reverse order of contact table]'
       '(-s)'{-s+,--sort=}'[sort contact table]:sort by:(first_name last_name)'
     )
     # search options
     local -a default_search_options merge_search_options
     default_search_options=(
-      '(-c)'{-c,--search-in-source-files}'[Look into source vcf files to speed up search queries in large address books]'
+      '(-c)'{-c,--search-in-source-files}'[look into source vcf files to speed up search queries in large address books]'
       '(-e)'{-e,--strict-search}'[narrow contact search to name field]'
       '(-u)'{-u+,--uid=}'[select contact by uid]:uid'
-      '*::search terms'
+      '*:: :_guard "^-*" "search term"'
```

Comment:
I would keep the description in plural form as the user can still give more than one argument as search terms.